### PR TITLE
fix channel dim selection on segmentation target

### LINKED ISSUE #1489: Semantic Segmentation target masks broken >0.7.5
## 🐛 Bug

The switch to albumentation in newer releases of lightning-flash seem to have broken transformation of segmentation targets.

This is what I expect masks to look like (screenshot showing below code sample running on 0.7.5):

2022-11-25 25-10-10-49--259_chrome

This is what it looks like on the latest release (0.8.1):

2022-11-25 25-13-13-32--468_chrome

### To Reproduce

Run the below sample with lightning-flash=0.7.5 and lightning-flash=0.8.1 and compare behavior. 

#### Code sample

```py
import torch

import flash
from flash.core.data.utils import download_data
from flash.image import SemanticSegmentation, SemanticSegmentationData

import matplotlib.pyplot as plt
import numpy as np

# 1. Create the DataModule
# The data was generated with the  CARLA self-driving simulator as part of the Kaggle Lyft Udacity Challenge.
# More info here: https://www.kaggle.com/kumaresanmanickavelu/lyft-udacity-challenge
# download_data(
#     "https://github.com/ongchinkiat/LyftPerceptionChallenge/releases/download/v0.1/carla-capture-20180513A.zip",
#     "./data",
# )

datamodule = SemanticSegmentationData.from_folders(
    train_folder="data/CameraRGB",
    train_target_folder="data/CameraSeg",
    val_split=0.1,
    transform_kwargs=dict(image_size=(256, 256)),
    num_classes=21,
    batch_size=4,
)

# 2. Build the task
model = SemanticSegmentation(
    backbone="mobilenetv3_large_100",
    head="fpn",
    num_classes=datamodule.num_classes,
)

n = 3
fig, axarr = plt.subplots(ncols=2, nrows=n, figsize=(8, 4*n))

for batch in datamodule.train_dataloader():
    print(batch.keys())
    for i in range(n):
        segm = batch['target'][i]
        print(segm.shape)
        img = np.rollaxis(batch['input'][i].numpy(), 0, 3)
        axarr[i, 0].imshow(img)
        axarr[i, 1].imshow(segm)
    break
```

### Environment

 - OS: Ubuntu WSL2
 - Python version: 3.10.8
 - GPU model: RTX 3080
 - CUDA Version: 11.6

### PR DESCRIPTION
**Pull request by izikgo**
## What does this PR do?

When loading mask files in semantic segmentation, the wrong axis is chosen causing a corrupted mask. The remaining pixels are then resized in the transformed function. This broke the examples in the semantic segmentation section.

Fixes #1489

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? [not needed for typos/docs]
- [x] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/image/segmentation/test_data.py [hunk 1]
```diff
@@ -58,6 +72,60 @@ def test_smoke():
...
+        os.makedirs(str(tmp_dir / "images"))
+        os.makedirs(str(tmp_dir / "targets"))
+
+        images = [
+            str(tmp_dir / "images" / "img1.png"),
+        ]
+
+        targets = [
+            str(tmp_dir / "targets" / "img1.png"),
+        ]
```
Comment (Borda) [inline tests/image/segmentation/test_data.py:101]:
```suggestion
        images = [
            str(tmp_dir / "images" / "img1.png")
        ]

        targets = [
            str(tmp_dir / "targets" / "img1.png")
        ]
```

**Review by Borda — APPROVED** [review]
